dev_requirements.txt

```diff
@@ -1,15 +1,5 @@
-black==22.12.0
-dbt-tests-adapter==1.4.5
-flake8>=3.5.0
-flaky>=3.5.3,<4
-freezegun==1.2.2
-ipdb
-isort
-mypy==1.1.1
-pre-commit
-pytest
-pytest-xdist>=2.1.0,<4
-pytz
-tox==3.27.1
-twine
-wheel
+dbt-tests-adapter~=1.4.5
+mypy==1.1.1  # patch updates have historically introduced breaking changes
+pre-commit~=3.2
+pytest~=7.2
+tox~=4.4
```

Comment:
why pin these?

Reply:
IMHO this is the best approach here.
It's better than pinning to a specific version like `tox==3.27.1`, because it allows us to always use the latest minor version without having to constantly update it.
It's also better than not specifying a version at all, since new major versions can potentially have breaking changes.
Additionally, this approach is used by `dbt-labs` in adapters that they maintain:
https://github.com/dbt-labs/dbt-snowflake/blob/main/dev-requirements.txt
WDYT?